portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,28 +14,58 @@
 
 package com.google.sps.servlets;
 
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.sps.data.Comment;
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
-import java.util.ArrayList;
 import com.google.gson.Gson;
+import java.util.ArrayList;
+import java.util.List;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-    private ArrayList<String> comments=new ArrayList<String>();
     @Override
     public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        PreparedQuery results = datastore.prepare(query);
+
+        List<Comment> comments = new ArrayList<>();
+        for (Entity entity : results.asIterable()) {
+            String name= (String) entity.getProperty("name");
+            String commentText = (String) entity.getProperty("commentText");
+            long timestamp = (long) entity.getProperty("timestamp");
+
+            Comment currcomment = new Comment(name, commentText, timestamp);
+            comments.add(currcomment);
+        }
+
         Gson gson = new Gson();
-        String json = gson.toJson(comments);
         response.setContentType("application/json;");
-        response.getWriter().println(json);
+        response.getWriter().println(gson.toJson(comments));
     }
     public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
         String newComment = request.getParameter("new-comment");
-        comments.add(0,newComment);
+        String newName = request.getParameter("new-name");
+        Entity commentEntity = new Entity("Comment");
+        
+        commentEntity.setProperty("name", newName);
+        commentEntity.setProperty("commentText", newComment);
+        commentEntity.setProperty("timestamp", System.currentTimeMillis());
+
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        datastore.put(commentEntity);
+
```

Comment:
I see that there are multiple temporary variables created in this method. Generally you should try to avoid temporary variables unless they are adding any additional info or unless the final statements to extract the values are too long. If the required value is buried deep down in the object Ex: object.methodA().methodB().methodC()...methodZ(), then we can try to break this statement to give more insights into what's happening here, otherwise we should always try to get rid of intermediate/short-lived variables.

Related article: https://techbeacon.com/app-dev-testing/why-unnecessary-variables-are-bad-your-code

Reply:
You can get rid of newComment, newName variables in this method.